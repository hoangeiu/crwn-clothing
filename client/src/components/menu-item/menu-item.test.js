import React from "react";
import { shallow } from "enzyme";

import { MenuItem } from "./menu-item.component";

describe("render MenuItem component", () => {
  let wrapper;
  let mockHistory;
  let mockMatch;
  let linkUrl = "/hats";
  beforeEach(() => {
    mockHistory = {
      push: jest.fn(),
    };
    mockMatch = {
      url: "/shop",
    };

    const mockProps = {
      title: "hats",
      imageUrl: "www.test.com",
      size: "large",
      history: mockHistory,
      linkUrl,
      match: mockMatch,
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  it("should MenuItem render without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push with the right string when menu-item is clicked", () => {
    wrapper.find(".menu-item").simulate("click");

    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
  });

  //   it("should pass size to MenuItemContainer as the prop size", () => {
  //     expect(wrapper.find("MenuItemContainer").prop("size")).toBe(size);
  //   });

  //   it("should pass imageUrl to BackgroundImageContainer as the prop imageUrl", () => {
  //     expect(wrapper.find("BackgroundImageContainer").prop("imageUrl")).toBe(
  //       imageUrl
  //     );
  //   });
});
