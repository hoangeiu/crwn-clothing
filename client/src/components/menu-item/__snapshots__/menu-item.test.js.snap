// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MenuItem component should MenuItem render without error 1`] = `
<div
  className="large menu-item"
  onClick={[Function]}
>
  <div
    className="background-image"
    style={
      Object {
        "backgroundImage": "url(www.test.com)",
      }
    }
  />
  <div
    className="content"
  >
    <h1
      className="title"
    >
      HATS
    </h1>
    <span
      className="subtitle"
    >
      SHOP NOW
    </span>
  </div>
</div>
`;
